core/src/miner/miner.rs

```diff
@@ -374,6 +374,11 @@ impl Miner {
         results
     }
 
+    pub fn delete_all_pending_transactions(&self) {
+        let mem_pool = &mut self.mem_pool.write();
```

Comment:
```rust
let mut mem_pool = self.mem_pool.write();
```

Reply:
The reason my (bad) code worked is that rust prolongs return value's lifetime when its reference is assigned directly with `let`, right? (I'll fix anyway)

Reply:
Yes, the lifetime of a temporary object assigned to a let is extended to the enclosing block.
You can read https://doc.rust-lang.org/reference/expressions.html#temporary-lifetimes for more details.
And actually, I don't think your code is bad. I believe that the compiler is smart enough to optimize this code. I just want to make the code clear.